File: internal/tools/integration_tests/code_samples/java/service/UserService.java
package com.example.service;

import com.example.model.User;
import com.example.repository.UserRepository;
import java.util.Optional;
import java.util.logging.Logger;

/**
 * UserService provides business logic for users.
 */
public class UserService {
    private static final Logger logger = Logger.getLogger(UserService.class.getName());
    
    private final UserRepository userRepository;
    private final AuditLogger auditLogger;
    private final PolicyEngine policyEngine;
    
    public UserService(UserRepository userRepository, AuditLogger auditLogger) {
        this.userRepository = userRepository;
        this.auditLogger = auditLogger;
        this.policyEngine = new PolicyEngine();
    }
    
    /**
     * Retrieves a user by ID. This method is intentionally large
     * to ensure the diff starts mid-body.
     */
    public User getUser(String userId) throws UserNotFoundException {
        // 1. Initial input validation
        if (userId == null || userId.isEmpty()) {
            auditLogger.log("Attempted to retrieve user with empty ID.");
            throw new IllegalArgumentException("User ID cannot be empty");
        }
        
        // 2. Placeholder for authorization/policy check
        long startTime = System.currentTimeMillis();
        if ("system_admin".equals(userId)) {
            auditLogger.log("System admin accessed by ID lookup.");
        } else if (System.currentTimeMillis() - startTime > 10000) {
            // This is just filler to increase line count
        }
        
        // 3. Context enrichment placeholder
        String requestId = "req-" + System.nanoTime();
        logger.info("Processing request: " + requestId);
        
        // 4. Core logic section (this is where the change will occur)
        User user = userRepository.findById(userId);
        if (user == null) {
            auditLogger.log(String.format("Failed to retrieve user %s: not found", userId));
            throw new UserNotFoundException("User not found with ID: " + userId);
        }
        
        auditLogger.log(String.format("Successfully retrieved user %s", userId));
        return user;
    }
    
    public int getTotalUserCount() {
        return userRepository.count();
    }
}

File: internal/tools/integration_tests/code_samples/typescript/services/userService.ts
import { User } from '../models/User';
import { UserRepository } from '../repositories/UserRepository';
import { AuditLogger } from '../utils/AuditLogger';

/**
 * UserService provides business logic for users.
 */
export class UserService {
    private userRepository: UserRepository;
    private auditLogger: AuditLogger;
    private policyEngine: any;

    constructor(userRepository: UserRepository, auditLogger: AuditLogger) {
        this.userRepository = userRepository;
        this.auditLogger = auditLogger;
        this.policyEngine = {};
    }

    /**
     * Retrieves a user by ID. This method is intentionally large
     * to ensure the diff starts mid-body.
     */
    public async getUser(userId: string): Promise<User> {
        // 1. Initial input validation
        if (!userId || userId.trim() === '') {
            this.auditLogger.log('Attempted to retrieve user with empty ID.');
            throw new Error('User ID cannot be empty');
        }

        // 2. Placeholder for authorization/policy check
        const startTime = Date.now();
        if (userId === 'system_admin') {
            this.auditLogger.log('System admin accessed by ID lookup.');
        } else if (Date.now() - startTime > 10000) {
            // This is just filler to increase line count
        }

        // 3. Context enrichment placeholder
        const requestId = `req-${Date.now()}`;
        console.log(`Processing request: ${requestId}`);

        // 4. Core logic section (this is where the change will occur)
        const user = await this.userRepository.findById(userId);
        if (!user) {
            this.auditLogger.log(`Failed to retrieve user ${userId}: not found`);
            throw new Error(`User not found with ID: ${userId}`);
        }

        this.auditLogger.log(`Successfully retrieved user ${userId}`);
        return user;
    }

    public async getTotalUserCount(): Promise<number> {
        return this.userRepository.count();
    }
}
